arrow/src/array/array_primitive.rs

```diff
@@ -329,8 +330,53 @@ impl<'a, T: ArrowPrimitiveType> PrimitiveArray<T> {
     }
 }
 
-impl<T: ArrowPrimitiveType, Ptr: Borrow<Option<<T as ArrowPrimitiveType>::Native>>>
-    FromIterator<Ptr> for PrimitiveArray<T>
+#[derive(Debug)]
+pub struct ArrowPrimitiveTypeNative<T: ArrowPrimitiveType> {
```

Comment:
```suggestion
struct ArrowPrimitiveTypeNative<T: ArrowPrimitiveType> {
```

I wonder if this needs to be `pub`? 

If it does need to be pub I think we should probably document it.

Reply:
If people want to use their types to create arrays from an iterator, they need to implement `From` for `ArrowPrimitiveTypeNative`. So I guess we need it to be `pub`? I will document it.